### ISSUE TITLE
Pasting into an empty block text area throws `Cannot apply deep operations on primitive value`

### ISSUE THREAD

**Issue by robertjanes** [sanity-io/sanity#4302]
**Describe the bug**

Pasting into an empty block text area throws the following error:
```
Error: Cannot apply deep operations on primitive values. Received patch with type "unset" and path "{"_key":"acf5fd264449"} that targeted the value "undefined"
    at _primitiveApply (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:144105:11)
    at _applyPatch (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:144144:10)
    at applyPatch5 (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:144147:15)
    at Array.reduce (<anonymous>)
    at applyAll2 (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:144132:18)
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:144175:22
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:152229:9
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:114111:7
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:114123:7
    at safelyCallDestroy (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-ZX3U2FOT.js?v=9f7159b1:23469:13)
```

**To Reproduce**

1. Create a schema which has an array field of block types. eg:
```
{
      name: 'text',
      title: 'Text',
      type: 'array',
      of: [
        {
          type: 'block',
        },
      ],
    },
```
2. Try and paste (CMD+v) into the **_empty_** text editor.

**Expected behavior**

The copied text is pasted.

**Which versions of Sanity are you using?**

3.7.1

**What operating system are you using?**

macOs Monterey 12.3

**Which versions of Node.js / npm are you running?**

8.19.2
v18.12.1

**Comment by frederikgdl** [comment]
I have the same issue. It seems to be working with version 3.7.0

**Comment by MichaelPrecel** [comment]
Hey @robertjanes 👋🦘. Was having the same issue – reverting to 3.7.0 also worked for me.

**Comment by georgebutter** [comment]
Hey all, obviously annoying, but you can type something into the field then paste and it works as a workaround.

**Comment by steveparrynz** [comment]
Same issue here.

**Comment by m-w-d-d** [comment]
Can confirm the same issue, pasting straight into a text block throws up that error, typing first and then paste no problems at all. 

Fresh install this morning.

**Comment by amcc** [comment]
same - also tried removing formatting on paste - i.e. cmd+shift+v on a mac

**Comment by archx3** [comment]
Same with me too .

**Comment by RobEasthope** [comment]
Reverting to sanity 3.7.0 hasn't solved it for me either. I've tried removing formatting via cmd+shift+v and and stuck the text through an online text formatter but to no effect. The standard `Rule.required()` validation is erroring out as well which matches up with the error below.

Pasting in a single paragraph and adding a carriage return yields the following error:
<img width="625" alt="image" src="https://user-images.githubusercontent.com/2979919/227788345-fec52409-38cd-4a78-b280-919bd466f6a5.png">

Typing in some text and then pasting in text as reported by @georgebutter and @m-w-d-d does work though so I'm using that as a work around for the time being. Thanks guys :)

**Comment by FavLucas** [comment]
Same issue for me too

**Comment by catherineriver** [comment]
Same for me. Downgrade to 3.7.0 solved issue.

**Comment by skogsmaskin** [comment]
> Thanks for the fix in 3.8.0 @skogsmaskin and the rest of the team

Oh, just noticed I forgot to update this issue (we have a couple of duplicates on this).

Yes this is now fixed. Closing this! Thank you for reporting and sorry about the inconvenience.

**Comment by robertjanes** [comment]
@skogsmaskin thanks for the quick fix!
